pre-commit.sh

```
@@ -0,0 +1,51 @@
+  echo "🔧 Running lint..."
+  npm run lint
+  
+  echo "🏗️  Running build..."
+  npm run build
+  
+  echo "🧪 Running tests..."
+  npm run test
+
+  # Adding formatted files to git stage.
+  echo "📝 Adding formatted files to git stage..."
+  git add $(git diff --name-only --cached --diff-filter=d | xargs)
```

Shellcheck (0.10.0)

[warning] Quote this to prevent word splitting.

(SC2046)